sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/DistributeUtils.java

```diff
@@ -47,15 +55,62 @@ class DistributeUtils {
      */
     static final String TESTER_APP_PACKAGE_NAME = "com.microsoft.hockeyapp.testerapp";
 
+    /**
+     * Notification channel identifier.
+     */
+    static final String NOTIFICATION_CHANNEL_ID = "appcenter.distribute";
+
     /**
      * Get the notification identifier for downloads.
      *
      * @return notification identifier for downloads.
      */
+    @VisibleForTesting
     static int getNotificationId() {
         return Distribute.class.getName().hashCode();
     }
 
+    static void postDownloadNotification(@NonNull Context context, String message, Intent intent) {
+        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+        Notification.Builder builder;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+
+            /* Create or update notification channel (mandatory on Android 8 target). */
+            NotificationChannel channel = new NotificationChannel(NOTIFICATION_CHANNEL_ID,
+                    context.getString(R.string.appcenter_distribute_notification_category),
+                    NotificationManager.IMPORTANCE_DEFAULT);
+            notificationManager.createNotificationChannel(channel);
+            builder = new Notification.Builder(context, NOTIFICATION_CHANNEL_ID);
+        } else {
+            builder = getOldNotificationBuilder(context);
+        }
+        int pendingIntentFlag = 0;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            pendingIntentFlag = PendingIntent.FLAG_IMMUTABLE;
+        }
+        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, pendingIntentFlag);
+        builder.setTicker(context.getString(R.string.appcenter_distribute_install_ready_title))
+                .setContentTitle(context.getString(R.string.appcenter_distribute_install_ready_title))
+                .setContentText(message)
+                .setSmallIcon(context.getApplicationInfo().icon)
+                .setStyle(new Notification.BigTextStyle().bigText(message))
+                .setContentIntent(pendingIntent);
+        Notification notification = builder.build();
+        notification.flags |= Notification.FLAG_AUTO_CANCEL;
+        notificationManager.notify(DistributeUtils.getNotificationId(), notification);
+    }
+
+    static void cancelDownloadNotification(@NonNull Context context) {
+        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+        notificationManager.cancel(DistributeUtils.getNotificationId());
+    }
+
+    @NonNull
+    @SuppressWarnings({"deprecation", "RedundantSuppression"})
+    private static Notification.Builder getOldNotificationBuilder(@NonNull Context context) {
```

Comment:
This "one-line" function is used in one place only. Do we need this function?

Reply:
Yes, to suppress warning